pkg/controller/master/node/node_down_controller.go

```diff
@@ -156,3 +174,89 @@ func getNodeCondition(conditions []corev1.NodeCondition, conditionType corev1.No
 	}
 	return cond
 }
+
+// deleteVolumeAttachmentsOfPod deletes all volume attachments of a pod,
+// when the pod is on a down node.
+// due to controller-manager handle too long time, cause VM restart wait a long time.
+// then we delete VolumeAttachments directly.
+// ref: https://github.com/harvester/harvester/issues/4049
+func (h *nodeDownHandler) deleteVolumeAttachmentsOfPod(pod *corev1.Pod) error {
+	volumeAttachments, err := h.getVolumeAttachmentsOfPod(pod)
+	if err != nil {
+		return err
+	}
+	for _, va := range volumeAttachments {
+		if va.DeletionTimestamp == nil {
+			err := h.vas.Delete(va.Name, &metav1.DeleteOptions{})
+			if err != nil {
+				if errors.IsNotFound(err) {
+					continue
+				}
+				return err
+			}
+			logrus.Infof("deleted volume attachment %v for pod %v on downed node %v",
+				va.Name,
+				pod.Name,
+				pod.Spec.NodeName)
+		}
+	}
+	return nil
+}
+
+func (h *nodeDownHandler) getVolumeAttachmentsOfPod(pod *corev1.Pod) ([]*storagev1.VolumeAttachment, error) {
+	res := make([]*storagev1.VolumeAttachment, 0)
+	volumeAttachments, err := h.vaCache.List(labels.Everything())
```

Comment:
Instead of listing all volumeAttachments in each pod iteration, how about getting all volumeAttachments first, and checking which one should be deleted?